File: Helpful_Maths.cpp
//only numbers to be summed = 1 , 2 , 3
//sum only of non-decreasing numbers
//just need to sort it

#include<bits/stdc++.h>

using namespace std;


int main(){
    string str;
    cin >> str;

    int count1 = 0 , count2 = 0 , count3 = 0;
    for(char c : str){
        if(c == '1'){
            count1 ++;
        }
        else if(c == '2'){
            count2 ++;
        }
        else if(c == '3'){
            count3 ++;
        }
    }

    // Print numbers with a plus sign between them
     // Print '1' count1 times
    for (int i = 0; i < count1; ++i) {
        cout << "1";
        if (i < count1 - 1 || count2 > 0 || count3 > 0) {
            cout << "+";
        }
    }

    // Print '2' count2 times
    for (int i = 0; i < count2; ++i) {
        cout << "2";
        if (i < count2 - 1 || count3 > 0) {
            cout << "+";
        }
    }

    // Print '3' count3 times
    for (int i = 0; i < count3; ++i) {
        cout << "3";
        if (i < count3 - 1) {
            cout << "+";
        }
    }
    
    cout << endl;
}

File: Helpful_Maths.py
str = input()

count1 = 0
count2 = 0
count3 = 0

for char in str:
    if char == '1':
        count1 += 1
    elif char == '2':
        count2 += 1
    elif char == '3':
        count3 += 1

answer = ""

if count1 > 0:
    for i in range(count1 - 1):
        answer += "1" + "+"
    if count2 != 0 or count3 != 0:
        answer += "1" + "+"
    else:
       answer += "1"

if count2 > 0:
    for i in range(count2 - 1):
        answer += "2" + "+"
    if count3 != 0:
        answer += "2" + "+"
    else:
        answer += "2"

if count3 > 0 :
    for i in range(count3-1):
        answer += "3" + "+"
    answer += "3"

print(answer)